tensorrt_llm/_torch/auto_deploy/transformations/transform.py

```diff
@@ -156,7 +156,7 @@ def __call__(self, cm: CachedSequenceInterface) -> GraphModule:
         ############################################################################################
 
         # load weights
-        self.factory.load_or_random_init(egm, device=cm.device)
+        self.factory.load_or_random_init(egm, device=self.factory.checkpoint_device or cm.device)
```

Comment:
```suggestion
        self.factory.load_or_random_init(egm, device=self.ad_config.checkpoint_device or cm.device)
```

If you do it like this you won't need to pass it through the factory interface and remove the `checkpoint_device` attribute in the factory class

Reply:
Thank you, made the update